src/nerpa_types.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import annotations
+from typing import (
+    Dict,
+    List,
+    Tuple)
+from enum import auto, Enum
+import yaml
+from dataclasses import asdict, dataclass
+
+LProb = float  # logarithm of probability
+MonomerResidue = str
+
+
+def enum_representer(dumper, e: Enum):
+    return dumper.represent_scalar(f'!{e.__class__.__name__}', e.name)
+
+
+class Chirality(Enum):
+    L = auto()
+    D = auto()
+    Unknown = auto()
+
+yaml.add_representer(Chirality, enum_representer)
+
+
+class PTM(Enum):  # post-translational modification
+    Methylation = auto()
+    Unknown = auto()
+
+yaml.add_representer(PTM, enum_representer)
+
+
+@dataclass
+class NRP_Monomer:
+    residue: MonomerResidue
+    ptms: Tuple[PTM, ...]
+    chirality: Chirality
+    rBAN_name: str
```

Comment:
Isn't this info redundant? I mean, can we always unequivocally deduce `rBAN_name` from `residue` (or vice versa)? If yes, probably it makes sense to have it rather as a function (here or in another Python file) than two fields in the class -- there is a high risk of filling at least one incorrectly when doing this manually. 

Of course, feel free to disagree, I might propose a stupid thing (here and in other places :) ).

Reply:
No. Say, rBAN name is `Cl-NMe-Arg`. Than residue will be `arg` and ptms will be (Unknown, Methylation). It is impossible to reconstruct rBAN name from residure and ptms.

Reply:
Ok, let's keep it as is then.